docs/madr/decisions/054-meshservice-identity-sync.md

```diff
@@ -0,0 +1,75 @@
+      app: redis
+      version: v1
+```
+
+We would require cert with tags `app=redis` and `version=v1`.
+However, there is no construct in Envoy that would verify SAN of `kuma://app/redis` **and** `kuma://version/v1`.
+We can only verify SAN 1 or SAN 2.
+Therefore, we need to "fallback" to `kuma.io/service`. 
+
+## Considered Options
+
+* Update MeshService with identity
+  * Store in `status`
+  * Store in `spec`
+
+## Decision Outcome
+
+Update MeshService with identity and store it in `spec`. 
+
+## Pros and Cons of the Options
+
+### Update MeshService with identity and sync it
+
+We need to have a place in MeshService to set an identity. We can specify a new field.
+
+```yaml
+spec: # or status, see discussion below
+  identities:
+  - type: ServiceTag
+    value: redis_kuma-demo_svc_80
+```
+
+Currently, the only type is `ServiceTag` but we want this API to be extensible to future changes of identity.
+
+This is a list because one MeshService can select multiple Dataplane proxies of multiple different `kuma.io/service`, so for mTLS we need to require either one of them.
+
+We need a component that updates this value in `MeshService` object.
+The component would go over all Dataplane objects, match proxies for MeshService and add all `kuma.io/service` tags to this list.
+Additionally unrelated to this MADR, we have to introduce a component that is equivalent of `ServiceInsight` data (count of offline, online dpps etc.) in Service status.
+This data also requires all matched Dataplane objects therefore it can be the same component.
+
+#### Spec vs Status
+
+We can place this information in `status` or in `spec`. The `status` field seems to be more natural place for it, because it's computed by CP.
+However, placing it in `status` comes with challenges.
+
+In multizone MADR we decided that we don't want to sync `status` from one zone to another because it's local to a source zone.
+If we were to place it in `status` we would have to add logic before syncing MeshService from global to zone to wipe out `status` but preserve this one field.
+While this might be reasonable, we need to additionally preserve other fields of status that were computed on the destination zone.
+
+Let's assume this situation:
+* Service A is synced from zone X to global
+* Service A is synced from global to zone Y
+* Zone Y updates status with VIP and a hostname
+* There is update to identities in zone X
+* Without any extra logic, KDS on Zone Y side would receive a resource, replace the MeshService including previously computed status. 
+  It would wipe out VIP and a hostname which will result in interruption of traffic.
+
+A simpler solution would be to put this info in `spec`. This way we have a single writer of `spec` (source zone) and single writer of `status` (destination zone).
+We need a new KDS flag `PreserveStatusOnZone`, which would indicate that whenever a Zone receives a resource it should preserve the existing `status`. 
```

Comment:
This implies we need to be careful in the MeshService reconciler to not lose these right?

Reply:
yes, but we are already quite careful in reconciler (I think current implem would not break this)